test/integration/integration_test.go

```diff
@@ -33,7 +33,7 @@ func TestKubeArchiveDeployments(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	retry.Do(func() error {
+	err = retry.Do(func() error {
```

Comment:
You should probably give this `err` variable a different name to distinguish it from the `err` used inside the loop. While the two won't collide programmatically, it is confusing for a reviewer. Especially since it is not referenced again until line 60, well below the function definition (so the conditional seems misplaced).

```suggestion
	retryErr = retry.Do(func() error {
```

Reply:
Changed!